modules/statistics/PreCallTest.ts

```diff
@@ -0,0 +1,48 @@
+import PreCallTest from '@jitsi/precall-test';
+
+
+export interface PreCallResult {
+    throughput: number; // Maximum bandwidth reached in kbps  (kilo bits per second).
+    fractionalLoss: number; // Packet loss percentage over all the test traffic.
+    rtt: number;  // Round trip time in milliseconds.
+    jitter: number; 
+    mediaConnectivity: boolean; // Whether the data channel was able to send data or not.
+}
+
+// Same interface as a PeerConnection configuration object.
+export interface IceServer {
+    urls: Array<string> | string;
+    username?: string;
+    credential?: string;
+}
+
+let preCallTest: any = null
+
+/**
+ * Run a pre-call test to check the network conditions. It uses a TURN server to establish
+ * a connection between two PeerConnections using the server as a relay. Afterwards it sends 
+ * some test traffic through a data channel to measure the network conditions, these are 
+ * recorded and returned through a Promise.
+ * 
+ * @param {Array<IceServer>} - The ICE servers to use for the test, these are passes to the PeerConnection constructor. 
+ * @returns {Promise<PreCallResult | string>} - A Promise that resolves with the test results or rejects with an error message.
+ */
+export default async function runPreCallTest(iceServers: Array<IceServer>): Promise<PreCallResult | string> {
+    // On initialization, the PreCallTest object simply does some checks and some browsers verifications,
```

Comment:
Can we have some static `init()` method perhaps? It could keep track of whether it has already run and avoid doing the work over and oer again.

Reply:
I meant the initial check we can reuse, not the whole test. This "init" thing would do just that.

Reply:
not sure I follow completely, you mean something like?
`const init = (function() = {
    preCallTest = new PreCallTest())
})()` And basically initialize on import? so I can remove the check from runPreCallTest?

Reply:
No, I mean something like:

```
await PrecallTest.globalInit();
const p = new PreCallTest();
...
```